### PR TITLE
Added option to split PDF into multiple parts and merge into one PDF

### PR DESCRIPTION
**Pull request by lm93129**
# Description

Added the option to merge into one PDF in the function of splitting a PDF into multiple parts to facilitate the work of splitting a single-page PDF in the middle and then merging it.

Closes #(issue_number)

## Checklist:

- [*] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [*] I have performed a self-review of my own code
- [*] I have commented my code, particularly in hard-to-understand areas
- [*] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Comment by sbplat** [comment]
What's the difference between this and the one from `MergeController`? Does it stitch together multiple pages into one (like 1x1+1x1=1x2)?

**Comment by lm93129** [comment]
For example, split a 10-page PDF horizontally and merge it into a 20-page PDF. (like 1x1+1x1=1x2+1x2)
Because for the convenience of printing, some PDFs will merge the original two pages of PDF into one page to facilitate cropping.

**Comment by sbplat** [comment]
Would splitting it into sections and then merging it using the other feature be the same?

**Comment by lm93129** [comment]
Yes, it's the same
For example
example.pdf：
page1: AB
page2: CD

Split Horizontally and merge：
example_output.pdf:
page1: A
page2: B
page3: C
page4: D

If not merged：
example_output.zip:
pdf1,pdf2,pdf3,pdf4

**Comment by lm93129** [comment]
When using horizontal splitting, multiple PDF files will be generated. This function only adds an option to directly merge them into a single PDF file after splitting.

The scenario is: in order to facilitate cropping after printing, the PDF exported by some systems will merge the document that should have been two pages into one page. This way you can fold it in half and cut it directly. However, for other processing needs, these PDF files need to be split horizontally and merged into one file. This feature also appears in other PDF tools.

**Comment by sbplat** [comment]
So this just simplifies a 2 step process into 1 step. I see what you mean now.

**Review by sbplat — COMMENTED** [review]
Can we reuse the implementation in `MergeController`?

File: src/main/java/stirling/software/SPDF/controller/api/SplitPdfBySectionsController.java [hunk 1]
```diff
@@ -145,4 +156,23 @@ public List<PDDocument> splitPdfPages(
...
+            doc.save(tempFile);
+            pdfMerger.addSource(tempFile);
+            tempFile.deleteOnExit();
+        }
+
+        pdfMerger.setDestinationStream(mergedPdfOutputStream);
+        pdfMerger.mergeDocuments(null);
+
+        return mergedPdfOutputStream;
+    }
```
Comment (sbplat) [inline src/main/java/stirling/software/SPDF/controller/api/SplitPdfBySectionsController.java]: Can we reuse the implementation in `MergeController`?

Comment (lm93129) [reply src/main/java/stirling/software/SPDF/controller/api/SplitPdfBySectionsController.java]: Okay, I will revise it again

File: src/main/java/stirling/software/SPDF/controller/api/SplitPdfBySectionsController.java [hunk 2]
```diff
@@ -53,8 +53,21 @@ public ResponseEntity<byte[]> splitPdf(@ModelAttribute SplitPdfBySectionsRequest
...
+        boolean merge = request.isMerge();
         List<PDDocument> splitDocuments = splitPdfPages(sourceDocument, verti, horiz);
+
+        String filename =
+                Filenames.toSimpleFileName(file.getOriginalFilename())
+                        .replaceFirst("[.][^.]+$", "");
+        if (merge) {
+            MergeController mergeController = new MergeController();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            mergeController.mergeDocuments(splitDocuments).save(baos);
```
Comment (sbplat) [inline src/main/java/stirling/software/SPDF/controller/api/SplitPdfBySectionsController.java:65]: Nice!

**Review by sbplat — APPROVED** [review]

**Comment by sbplat** [comment]
Thanks!

**Pull request merged**